build-tools/packages/build-cli/src/typeValidator/typeValidatorConfig.ts

```diff
@@ -18,18 +18,39 @@ export interface BrokenCompatSettings {
  */
 export type BrokenCompatTypes = Partial<Record<string, BrokenCompatSettings>>;
 
+// Duplicate of the ApiLevel type defined in build-cli/src/library/apiLevel.ts
+// AB#12469 tracks moving the type test infra into build-cli, at which point this duplicate type won't be needed.
+export type ApiLevel = "public" | "beta" | "alpha" | "internal" | "legacy";
+
 export interface ITypeValidationConfig {
+	/**
+	 * The entrypoint (API level) for which type tests should be generated. This value can be overridden when using
+	 * `flub generate typetests` by passing the `--entrypoint` flag. If this value is not provided, it will default to
+	 * {@link ApiLevel.legacy}.
+	 *
+	 * @defaultValue {@link ApiLevel.legacy}
+	 */
+	entrypoint: ApiLevel;
+
 	/**
 	 * An object containing types that are known to be broken.
 	 */
 	broken: BrokenCompatTypes;
 
 	/**
 	 * If true, disables type test preparation and generation for the package.
+	 *
+	 * @defaultValue `false`
 	 */
 	disabled?: boolean;
 }
 
+export const defaultTypeValidationConfig: ITypeValidationConfig = {
+	entrypoint: "legacy",
```

Comment:
Should "public" be the default? Not that avoiding breaking changes in legacy isn't important, but it seems like avoiding them in public should be _the_ top priority?

Reply:
Legacy is a superset of public (I think - I need to double-check with Jason). Public covers the least amount of API surface.

Reply:
legacy is indeed a superset of public (really for consumer convenience).
alpha is super set of beta and beta is superset of public as well.
alpha+beta+legacy only intersect with what is public. So, legacy does not include alpha or beta.

Reply:
Thanks for clarifying, Jason. Regardless of what we pick as the "default" here in code, ultimately each package explicitly states its own default, and I am planning to persist that in the package.json (explicit is better than implicit).

The current design proposal is that if we need to test legacy and alpha/beta, we generate multiple type test files by running the generator multiple times for each entrypoint we care about. There are alternatives we could consider if that proves too onerous.

Reply:
If we want things to be explicit, then no default should be an option on the table.
I think oclif allows checks to say --entrypoint is required if --level isn't used.